src/sentry/static/sentry/app/views/performance/traceDetails/transactionBar.tsx

```diff
@@ -223,11 +231,47 @@ class TransactionBar extends React.Component<Props> {
     );
   }
 
+  renderRectangle() {
+    const {transaction, traceInfo, theme} = this.props;
+
+    const palette = theme.charts.getColorPalette(traceInfo.maxGeneration);
+
+    // TODO(tonyx) Does this need to handle any unexpected data? (e.g. start > end)
```

Comment:
I don't think we need to put any defensive frontend code here, endTimestamp being greater than start is something we really should be fixing on the backend, with no frontend handling.

Reply:
Hmm should the backend just skip transactions that have bad timestamps?

Reply:
can we somehow flag that the timestamps might be bad and call it out in a warning instead?

Reply:
This could be possible because Relay doesn't normalize span timestamps. You can try to do this by manually sending a transaction event with reversed timestmaps.